bibtex-actions.el

```diff
@@ -611,35 +607,25 @@ interactively when `bibtex-actions-at-point-fallback' is non-nil.
 With prefix ARG, rebuild the cache before offering candidates."
   (interactive "P")
   (if (fboundp 'embark-dwim)
-      (let ((embark-keymap-alist '((bibtex . bibtex-actions-map))))
-        (condition-case err
-            (let ((embark-target-finders '(bibtex-actions-citation-key-at-point))
-                  (embark-keymap-alist '((citation-key . bibtex-actions-buffer-map)))
-                  ;; `embark-general-map' is useless in this case
-                  ;; because we are not in the minibuffer and the
-                  ;; TARGET is not string
-                  (embark-general-map embark-meta-map))
-              ;; This is a workaround: unless in the minibuffer,
-              ;; embark cannot pass TARGET to action when the action
-              ;; is an interactive command
-              (cl-letf (((symbol-function 'embark--act)
-                         (lambda (action target _)
-                           (if (where-is-internal action (list bibtex-actions-buffer-map))
-                               (funcall action target)
-                             (command-execute action)))))
-                (if bibtex-actions-embark-dwim
-                    (embark-dwim)
-                  (embark-act))))
-          (user-error
-           (when (and (string-equal (error-message-string err) "No target found")
-                      bibtex-actions-at-point-fallback)
-             (bibtex-actions-run-default-action
-              (bibtex-actions-read :rebuild-cache arg))))))
+      (condition-case err
+          (if bibtex-actions-embark-dwim
+              (embark-dwim)
+            (embark-act))
+        (user-error
+         (when (and (string-equal (error-message-string err) "No target found")
+                    bibtex-actions-at-point-fallback)
+           (bibtex-actions-run-default-action
+            (bibtex-actions-read :rebuild-cache arg)))))
     (if-let ((keys (bibtex-actions-citation-key-at-point)))
         (funcall bibtex-actions-default-action keys)
       (when bibtex-actions-at-point-fallback
         (bibtex-actions-run-default-action
          (bibtex-actions-read :rebuild-cache arg))))))
 
+(with-eval-after-load "embark"
+  (add-to-list 'embark-target-finders 'bibtex-actions-citation-key-at-point)
+  (add-to-list 'embark-keymap-alist '((bibtex . bibtex-actions-map)
+                                      (citation-key . bibtex-actions-buffer-map))))
```

Comment:
```suggestion
(with-eval-after-load "embark"
  (add-to-list 'embark-target-finders 'bibtex-actions-citation-key-at-point)
  (add-to-list 'embark-keymap-alist '(bibtex . bibtex-actions-map))
  (add-to-list 'embark-keymap-alist '(citation-key . bibtex-actions-buffer-map)))
```

Reply:
I think this should solve the problem that `embark-act` doesn't work.

Reply:
Indeed, the add-to-list statement was wrong.